File: lib/problem1.rb
class VendingMachine
  attr_reader :bottles
  def initialize(bottles = 20)
    @bottles = bottles
  end
  def purchase(amount)
    @bottles -= amount
  end

  def restock(amount)
    @bottles += amount
  end
  def get_inventory
      puts @bottles
  end
  def report
      puts "Inventory: #{@bottles} bottles"
  end
end

bottleorder = VendingMachine.new(20)
bottleorder.purchase(5)
bottleorder.restock(2)
bottleorder.get_inventory()
bottleorder.report()

File: problem1.py
class VendingMachine:
    def __init__(self):
        self.bottles = 20

    def purchase(self, amount):
        self.bottles = self.bottles - amount

    def restock(self, amount):
        self.bottles = self.bottles + amount

    def get_inventory(self):
        return self.bottles

    def report(self):
        print(f"Inventory: {self.bottles} bottles")


if __name__ == "__main__":
    drink_machine = VendingMachine()
    drinks_to_buy = int(input(f"amount to buy: "))
    bottles_to_stock = int(input(f"amount to restock: "))
    drink_machine.purchase(drinks_to_buy)
    drink_machine.restock(bottles_to_stock)
    drink_machine.report()